client/src/components/Header/components/nav-links.tsx

```diff
@@ -64,23 +86,198 @@ export class NavLinks extends Component<NavLinksProps, {}> {
     );
   }
 
-  handleLanguageChange = (
-    event: React.ChangeEvent<HTMLSelectElement>
-  ): void => {
-    const { toggleDisplayMenu, navigate } = this.props;
-    toggleDisplayMenu();
+  getPreviousMenuItem(target: HTMLElement): HTMLElement {
+    const { menuButtonRef } = this.props;
+    const currentListItem = target.closest('.nav-list > li');
+    if (!currentListItem) {
+      return menuButtonRef.current;
+    }
+    try {
+      const previousSibling =
+        currentListItem.previousSibling as HTMLElement | null;
+      const previousListItem = previousSibling.querySelector('a, button');
+      return previousListItem;
+    } catch (e) {
+      return menuButtonRef.current;
+    }
```

Comment:
Why is this a try catch block?  It seems like this is using error handling to handle normal behaviour.